x/ibc/07-tendermint/misbehaviour_test.go

```diff
@@ -45,76 +45,122 @@ func (suite *TendermintTestSuite) TestCheckMisbehaviour() {
 	testCases := []struct {
 		name            string
 		clientState     clientexported.ClientState
-		consensusState  clientexported.ConsensusState
+		consensusState1 clientexported.ConsensusState
+		consensusState2 clientexported.ConsensusState
 		evidence        clientexported.Misbehaviour
 		consensusParams *abci.ConsensusParams
-		height          uint64
 		timestamp       time.Time
 		expPass         bool
 	}{
 		{
 			"valid misbehavior evidence",
 			types.NewClientState(chainID, types.DefaultTrustLevel, trustingPeriod, ubdPeriod, maxClockDrift, height, commitmenttypes.GetSDKSpecs()),
 			types.NewConsensusState(suite.now, commitmenttypes.NewMerkleRoot(tmhash.Sum([]byte("app_hash"))), height, bothValsHash),
+			types.NewConsensusState(suite.now, commitmenttypes.NewMerkleRoot(tmhash.Sum([]byte("app_hash"))), height, bothValsHash),
 			types.Evidence{
 				Header1:  types.CreateTestHeader(chainID, height, height, suite.now, bothValSet, bothValSet, bothSigners),
 				Header2:  types.CreateTestHeader(chainID, height, height, suite.now.Add(time.Minute), bothValSet, bothValSet, bothSigners),
 				ChainID:  chainID,
 				ClientID: chainID,
 			},
 			simapp.DefaultConsensusParams,
-			height,
 			suite.now,
 			true,
 		},
 		{
 			"valid misbehavior at height greater than last consensusState",
 			types.NewClientState(chainID, types.DefaultTrustLevel, trustingPeriod, ubdPeriod, maxClockDrift, height, commitmenttypes.GetSDKSpecs()),
 			types.NewConsensusState(suite.now, commitmenttypes.NewMerkleRoot(tmhash.Sum([]byte("app_hash"))), height-1, bothValsHash),
+			types.NewConsensusState(suite.now, commitmenttypes.NewMerkleRoot(tmhash.Sum([]byte("app_hash"))), height-1, bothValsHash),
 			types.Evidence{
 				Header1:  types.CreateTestHeader(chainID, height, height-1, suite.now, bothValSet, bothValSet, bothSigners),
 				Header2:  types.CreateTestHeader(chainID, height, height-1, suite.now.Add(time.Minute), bothValSet, bothValSet, bothSigners),
 				ChainID:  chainID,
 				ClientID: chainID,
 			},
 			simapp.DefaultConsensusParams,
-			height - 1,
+			suite.now,
+			true,
+		},
+		{
+			"valid misbehavior evidence with different trusted heights",
+			types.NewClientState(chainID, types.DefaultTrustLevel, trustingPeriod, ubdPeriod, maxClockDrift, height, commitmenttypes.GetSDKSpecs()),
+			types.NewConsensusState(suite.now, commitmenttypes.NewMerkleRoot(tmhash.Sum([]byte("app_hash"))), height-1, bothValsHash),
```

Comment:
nit: use vars `trustedHeight1` and `trustedHeight2`, much easier to read to ensure correctness

Reply:
`trustedHeight1, trustedHeight2 := height-1, height -3`

Reply:
Hard to do this since this is being done directly in struct initialization. I could try to define these above the test cases, but where they get used is intermingled enough that it would be confusing